docs/adr/0007-azure-pipeline-security-model.md

```diff
@@ -0,0 +1,70 @@
+# Azure pipeline security model
+
+## Status
+
+Proposed
+
+
+## Context
+
+This ADR defines an approach for a least-privilege Azure DevOps Pipeline that supports the automated deployment, teardown and re-deployment of a complete Marain instance.
+
+The challenge here is to balance the varied benefits of having a self-sufficient, repeatable pipeline with the need for adequate security controls to minimise the risk of the pipeline becoming the source of a security incident.
+
+Such a deployment pipeline needs to be able to perform the following actions:
+
+1. Create resources in the target subscription
+1. Query AAD for existing applications & service principals
+1. Perform ARM role assignments in the target subscription
+1. Create and manage new AAD applications & service principals
+
+### Create resources in the target subscription
+By default, the 'Service Connections' that Azure Pipelines use to authenticate with Azure only provide #1 of the above.
+
+### Perform ARM role assignments in the target subscription
+Typically this is achieved by granting the identity behind the service connection the 'owner' role in the target subscription.
```

Comment:
I _thought_ you could do this by making it the owner just of the relevant resource group.

There's a bootstrap problem there of course. The resource group has to exist before you can make the service principal the owner. Making it the owner of the entire sub makes that problem go away, but this is not a "least privilege" solution.

Reply:
Ah...I've realised after reading on that you're describing what it "typically" done, but this isn't actually what you're proposed.

Consider adding more text to this saying what the alternatives are.